src/jit/lowerxarch.cpp

```diff
@@ -2366,104 +2478,468 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
     HWIntrinsicCategory category    = Compiler::categoryOfHWIntrinsic(intrinsicID);
     HWIntrinsicFlag     flags       = Compiler::flagsOfHWIntrinsic(intrinsicID);
     int                 numArgs     = Compiler::numArgsOfHWIntrinsic(node);
-    GenTree*            op1         = node->gtGetOp1();
-    GenTree*            op2         = node->gtGetOp2();
+    var_types           baseType    = node->gtSIMDBaseType;
+
+    GenTree* op1 = node->gtGetOp1();
+    GenTree* op2 = node->gtGetOp2();
+    GenTree* op3 = nullptr;
 
     if ((flags & HW_Flag_NoContainment) != 0)
     {
+#if DEBUG
+        // In DEBUG only, validate that we are only handling
+        // intrinsics we've validated should go down this code path.
+
+        switch (intrinsicID)
+        {
+            // TODO-XArch-CQ: A number of these intrinsics do support containment, but
+            //                may require some more complicated changes to support.
+
+            // TODO-XArch-CQ: Some of these nodes are also containable. For example,
+            //                ConvertToSingle exists purely to satisfy the type system
+            //                and can be directly consumed by various other intrinsics.
+
+            case NI_SSE_CompareEqualOrderedScalar:
```

Comment:
I'm not sure I see the value in these huge switches, as it is basically duplicate info that's in the table. Can you elucidate?

Reply:
@CarolEidt, there is a lot of data in the tables, and in many cases, I've found various data to be wrong/incorrect (there are a few fixes in this PR even).

These switches help validate that someone has reviewed the intrinsic and ensured that it should in-fact be handled here (rather than letting new or existing intrinsics silently go down the wrong path and potentially cause code-breakages).

Reply:
After some more thought, I removed the switch tables.

Ideally, we would be able to write unit tests to validate all this, but that isn't readily possible today.

I think we might be able to come up with something better to help validate that the table is correct, and stays correct, however.

Reply:
@tannergooding  - thanks. Would you like to submit an issue (or is there already one) suggesting that we invest in unit testing for the HW intrinsics?